src/main/xerus/monstercat/api/Covers.kt

```diff
@@ -15,7 +15,7 @@ object Covers {
 	
 	private fun coverCacheFile(coverUrl: String, size: Int): File {
 		coverCacheDir.mkdirs()
-		val newFile = coverCacheDir.resolve(coverUrl.substringAfterLast('/').replaceIllegalFileChars())
+		val newFile = coverCacheDir.resolve(coverUrl.substringBeforeLast('/').substringAfterLast('/').replaceIllegalFileChars())
```

Comment:
huh?

Reply:
New cover url is connect.monstercat.com/v2/release/[release ID]/cover so the old code saved every cover under the same filename "cover". This now saves them under the release ID.
That makes me think that I need to use the same code as in the downloader to get the MIME type and add an extension like .png

Reply:
yeah, I was also wondering why this function duplicates the functionality in the downloader

Reply:
This is a story for another day. We should focus on fixing the already existing code, I'll make another commit afterwards to refactor those functions

Reply:
sounds good